setup.py

```diff
@@ -39,7 +39,7 @@ def get_install_requires() -> List[str]:
         "scipy!=1.4.0" if sys.version[:3] == "3.6" else "scipy>=1.7.0",
         "sqlalchemy>=1.1.0",
         "tqdm",
-        "typing_extensions>=3.10.0.0",
+        "typing_extensions",
```

Comment:
We should set the lower limit actually. I'll work on it at #3974.

Reply:
[Comment] Do we really need the lower limit? I think this dependency is optional and we could leave the lower limit unspecified.

Reply:
Thank you for your comment. You are right. We can move typing_extensions to optional dependencies like #3990.